src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -660,41 +706,84 @@ private void initializeInputPreloadExecutorService() {
         return chromosomeIntervals.stream().map(interval -> {
-            GenomicsDBImportConfiguration.Partition.Builder partitionBuilder = GenomicsDBImportConfiguration.Partition.newBuilder();
             Coordinates.ContigPosition.Builder contigPositionBuilder = Coordinates.ContigPosition.newBuilder();
-            Coordinates.GenomicsDBColumn.Builder columnBuilder = Coordinates.GenomicsDBColumn.newBuilder();
+            Coordinates.GenomicsDBColumn.Builder beginBuilder = Coordinates.GenomicsDBColumn.newBuilder();
+            Coordinates.GenomicsDBColumn.Builder endBuilder = Coordinates.GenomicsDBColumn.newBuilder();
             //begin
             contigPositionBuilder.setContig(interval.getContig()).setPosition(interval.getStart());
-            columnBuilder.setContigPosition(contigPositionBuilder.build());
-            partitionBuilder.setBegin(columnBuilder.build());
+            beginBuilder.setContigPosition(contigPositionBuilder.build());
             //end
             contigPositionBuilder.setPosition(interval.getEnd());
-            columnBuilder.setContigPosition(contigPositionBuilder.build());
-            partitionBuilder.setEnd(columnBuilder.build());
-            partitionBuilder.setWorkspace(workspace);
-            partitionBuilder.setGenerateArrayNameFromPartitionBounds(true);
-            return partitionBuilder.build();
+            endBuilder.setContigPosition(contigPositionBuilder.build());
+            return createPartitionWithBeginAndEnd(beginBuilder.build(), endBuilder.build());
         }).collect(Collectors.toList());
     }
 
+    private void generateIntervalListFromVidMap() {
+        try {
+            GenomicsDBVidMapProto.VidMappingPB vidMapPB = 
+                org.broadinstitute.hellbender.tools.genomicsdb.GenomicsDBUtils.getProtobufVidMappingFromJsonFile(vidMapJSONFile);
+    
+            String[] partitions = GenomicsDBUtils.listGenomicsDBArrays(workspace);
+            intervals = new ArrayList<>();
+            for (int i=0; i<partitions.length; i++) {
+                long[] bounds = GenomicsDBUtils.getArrayColumnBounds(workspace, partitions[i]);
+                // merge-contigs-into-num-partitions ensures entire contigs are within a given partition
+                // so we just check here that contig starts within the given bounds
+                intervals.addAll(vidMapPB.getContigsList().stream()
+                        .filter(x -> x.getTiledbColumnOffset() >= bounds[0] &&  
+                        x.getTiledbColumnOffset() <= bounds[1])
+                        .map(x -> new SimpleInterval(x.getName(), 1, Math.toIntExact(x.getLength())))
+                        .collect(Collectors.toList()));
+            }
+        } catch (final IOException e) {
+            throw new UserException("Could not get vid map protobuf from file:" + vidMapJSONFile + 
+                    ". Is the workspace corrupted?");
+        }
+    }
+
     private void generateIntervalListFromWorkspace() {
```

Comment:
Similarly, this seems like it could return the value instead of modifying the field.

Reply:
done